### PR TITLE
Dependency Injection

### PR DESCRIPTION
**Pull request by MigleMic**
After lecture, had to change my idea a bit because lecturer said messages do not fill the requirenment

Added IUser interface and now all actions that have to use user, need to be used with UserAccess instead of DatabaseAccess.
I have commented the code in DatabaseAccess which are connected with user database in case we will want to move back to it.

### CODE REVIEW

File: Encountify/Services/ScoreboardCreation.cs [hunk 1]
```diff
@@ -11,10 +12,9 @@ public class ScoreboardCreation
         public ScoreboardEntry this[int i] => CreateScoreboard().ToArray()[i];
         public List<ScoreboardEntry> CreateScoreboard(bool reversed = false)
         {
-            DatabaseAccess<User> userData = new DatabaseAccess<User>();
+            IUser userData = DependencyService.Get<IUser>();    //new DatabaseAccess<User>();
             List<User> users = (List<User>)userData.GetAllAsync().Result;
-            DatabaseAccess<VisitedLocations> visitedLocationsData = new DatabaseAccess<VisitedLocations>();
-            List<VisitedLocations> visitedLocations = (List<VisitedLocations>)visitedLocationsData.GetAllAsync().Result;
+            DatabaseAccess<VisitedLocations> visitedLocationsData = new DatabaseAccess<VisitedLocations>(); List<VisitedLocations> visitedLocations = (List<VisitedLocations>)visitedLocationsData.GetAllAsync().Result;
```
Comment (Waltzerizm) [inline Encountify/Services/ScoreboardCreation.cs:17]: Very long snake.
"Enter" would be nice between those variables LOL

**Review by Waltzerizm — APPROVED** [review]
Looks good! Not sure if it works fine but will find out when I will be using it:-)

**Review by Mdominykas — APPROVED** [review]
Looks good and works

File: Encountify/Services/ScoreboardCreation.cs (hunk 1 above)
Comment (MigleMic) [reply Encountify/Services/ScoreboardCreation.cs]: Thank you for pointing out, haven't noticed it

**Pull request merged**